mysql/Doc_Office.sql

```diff
@@ -1,87 +1,182 @@
-    Room_number int,
-    PRIMARY KEY (Appointment_number),
-    FOREIGN KEY (Patient_SSN) REFERENCES PATIENT(SSN),
-    FOREIGN KEY (Test_given) REFERENCES MEDICAL_TEST(Test_ID),
-    FOREIGN KEY (Doctor_ID) REFERENCES DOCTOR(Doctor_ID)
+DROP TABLE IF EXISTS SPECIALIZATION;
+CREATE TABLE SPECIALIZATION(
+	specialtyid int,
+    specialty varchar(45),
+    PRIMARY KEY (specialtyid)
     );
-
+    
 DROP TABLE IF EXISTS AUDIT;
 CREATE TABLE AUDIT(
-	Doctor_ID int AUTO_INCREMENT,
+	id int AUTO_INCREMENT,
+	Doctor_ID int,
     Action varchar(100),
     Date_modified datetime,
+    PRIMARY KEY (id),
     FOREIGN KEY (Doctor_ID) REFERENCES DOCTOR(Doctor_ID)
     );
+
+DROP TABLE IF EXISTS DOCTORPATIENT;
+CREATE TABLE DOCTORPATIENT (
+  doctorId INT NOT NULL,
+  patientid char(9) NOT NULL,
+  PRIMARY KEY (doctorId, patientid),
+  CONSTRAINT doctorid
+    FOREIGN KEY (doctorId)
+    REFERENCES DOCTOR (Doctor_ID)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT patientid
+    FOREIGN KEY (patientid)
+    REFERENCES PATIENT (SSN)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION
+);
+
+DROP TABLE IF EXISTS DOCTORTEST;
+CREATE TABLE DOCTORTEST(
+	doctorId int NOT NULL,
+    testid int NOT NULL,
+    PRIMARY KEY (doctorId, testid),
+    CONSTRAINT doctor_id
+	  FOREIGN KEY (doctorId) REFERENCES DOCTOR(Doctor_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT testid
+      FOREIGN KEY (testid) REFERENCES TEST(Test_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+
+DROP TABLE IF EXISTS DOCTORSPECIALIZATION;
+CREATE TABLE  DOCTORSPECIALIZATION(
+	doctorId int NOT NULL,
+    specialtyid int NOT NULL,
+    PRIMARY KEY (doctorId, specialtyid),
+    CONSTRAINT doctor
+	  FOREIGN KEY (doctorId) REFERENCES DOCTOR(Doctor_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT specialtyid
+      FOREIGN KEY (specialtyid) REFERENCES SPECIALIZATION(specialtyid)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+
+DROP TABLE IF EXISTS TESTAPPOINTMENT;
+CREATE TABLE  TESTAPPOINTMENT(
+	testid int NOT NULL,
+    appointmentid int NOT NULL,
+    PRIMARY KEY (testid, appointmentid),
+    CONSTRAINT test_id
+	  FOREIGN KEY (testid) REFERENCES TEST(Test_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT appointmentid_
+      FOREIGN KEY (appointmentid) REFERENCES APPOINTMENT(Appointment_number)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+    
+DROP TABLE IF EXISTS TESTPATIENT;
+CREATE TABLE  TESTPATIENT(
+	testid int NOT NULL,
+	patientid char(9) NOT NULL,
+    PRIMARY KEY (testid, patientid),
+    CONSTRAINT testid_
+	  FOREIGN KEY (testid) REFERENCES TEST(Test_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT patientid_
+      FOREIGN KEY (patientid) REFERENCES PATIENT(SSN)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+    
+DROP TABLE IF EXISTS PRESCRIPTIONPATIENT
+CREATE TABLE PRESCRIPTIONPATIENT(
```

Comment:
I don't think we need this, we can just have each prescription entry have a patientSSN since each prescription should be unique. We could use this if we had a table of drugs to give out but we're not doing that